doc/api/errors.md

```diff
@@ -573,6 +580,79 @@ an argument of the wrong type has been passed to a Node.js API.
 The `'ERR_INVALID_CALLBACK'` error code is used generically to identify that
 a callback function is required and has not been provided to a Node.js API.
 
+<a id="ERR_INVALID_FILE_URL_HOST"></a>
+### ERR_INVALID_FILE_URL_HOST
+
+An error with the `'ERR_INVALID_FILE_URL_HOST'` code may be thrown when a
+Node.js API that consumes `file:` URLs (such as certain functions in the
+[`fs`][] module) encounters a file URL with an incompatible host. Currently,
+this situation can only occur on Unix-like systems, where only `localhost` or
+an empty host is supported.
+
+<a id="ERR_INVALID_FILE_URL_PATH"></a>
+### ERR_INVALID_FILE_URL_PATH
+
+An error with the `'ERR_INVALID_FILE_URL_PATH'` code may be thrown when a
+Node.js API that consumes `file:` URLs (such as certain functions in the
+[`fs`][] module) encounters a file URL with an incompatible path. The exact
+semantics for determining whether a path can be used is platform-dependent.
+
+<a id="ERR_INVALID_THIS"></a>
+### ERR_INVALID_THIS
+
+The `'ERR_INVALID_THIS'` error code is used generically to identify that a
+Node.js API function is called with an incompatible `this` value. Its use is
+currently limited to the [WHATWG URL API][] to satisfy specification
+requirements, but may be expanded to other areas of the Node.js API in the
+future.
+
+Example:
+
+```js
+const { URLSearchParams } = require('url');
+const urlSearchParams = new URLSearchParams('foo=bar&baz=new');
+
+const buf = Buffer.alloc(1);
+urlSearchParams.has.call(buf, 'foo');
+  // Throws a TypeError with code 'ERR_INVALID_THIS'
+```
+
+<a id="ERR_INVALID_TUPLE"></a>
+### ERR_INVALID_TUPLE
+
+An error with code `'ERR_INVALID_TUPLE'` is thrown when an element in the
+`iterable` provided to the [WHATWG][WHATWG URL API] [`URLSearchParams`
+constructor][`new URLSearchParams(iterable)`] does not represent a name/value
+tuple -- that is, if an element is not iterable, or does not consist of exactly
+two elements.
+
+<a id="ERR_INVALID_URL"></a>
+### ERR_INVALID_URL
+
+An error using the `'ERR_INVALID_URL'` code is thrown when an invalid URL is
+passed to the [WHATWG][WHATWG URL API] [`URL` constructor][`new URL(input)`] to
+be parsed. The thrown error object typically has an additional property
+`'input'` that contains the URL that failed to parse.
```

Comment:
Right, the `REPLACE_INVALID_UTF8` is pretty edge-case, but trimming can indeed be something the user does not expect. Agreed on addressing this in a separate PR though.